### DISCUSSION TITLE
SHAP values in binary classification are additive inverses, why?

### DISCUSSION THREAD

**Discussion by iteimouri** [shap/shap#1619] [General]
I am going through [this turorial](https://medium.com/analytics-vidhya/shap-part-3-tree-shap-3af9bcd7cd9b). This is a binary classification problem, the author notes that for binary classification the SHAP value is additive inverses. In other words, for each feature there will be an equal contribution to each class. This becomes more clear if we look at the summary plot: 

![](https://miro.medium.com/max/2000/1*2W2PAed0qDltlCqvn21j9w.png)

You can see that for each feature there is equal blue and red (contribution to each class). I have reproduced the same result for another binary problem. My explanation for this is that the SHAP value which is calculated for each feature in a binary classification does not have any mixing term and hence the result would only be symmetrical. I would however like to know the exact mathematical formulation for this if anyone knows or can lead me to a source?

**Comment by letsdecker** [comment]
Nobody knows? I have the same problem. Did you figure out why did you have this output?

**Comment by CloseChoice** [comment]
The reason for this is that shap values only show the contribution ABOVE baseline. Imagine if 60% of your samples belong to class 0, so this is the baseline. Let's assume observation X has a probability of 70% to belong to class 0 according to our classifier. That means that something increased the probability of observation X about 10% above baseline to belong to class 0 (or decreased the probability about 10% to belong to class 1). You see that any change in probabilities is additive inverse if taken relative to the baseline
